### ISSUE TITLE
I2S and DMA support for ESP32

### ISSUE THREAD

**Issue by costyn** [wokwi/wokwi-features#213]
**Is your feature request related to a problem? Please describe.**
It would be cool if we can use the FastLED clockless I2S/DMA driver on ESP32, so we can test it against "real world" code.

Also, since it seems the FastLED on ESP32 with RMT is broken, this might be a nice alternative.

**Describe the solution you'd like**
I would like if my I2S test project works.
https://wokwi.com/arduino/projects/312993650809242176

To use the native FastLED I2S driver, add `#define FASTLED_ESP32_I2S true` to the top of your sketch:
https://wokwi.com/arduino/projects/312460386125218368

**Comment by urish** [comment]
This issue has been getting a lot of votes recently, and I'd love to learn why people are voting for it.

Do you need I2S on the ESP32 for:

1. Audio output
2. Audio input
3. FastLED
4. Other purpose (please specify)

Please add your comments to this discussion - they'll help prioritizing our work.

**Comment by costyn** [comment]
I have been using my vote-powers on this issue since I first got them :) Not sure who else has been voting, but I need it for FastLED. I don't need it very badly, but I don't my vote-powers for anything else.

**Comment by urish** [comment]
> but I need it for FastLED

Can you share a link to a Wokwi project that has the setup you'd like to use? (FastLED + I2S)?

**Comment by costyn** [comment]
https://wokwi.com/projects/315542185092055618  For example. The real-world project using part of these algorithms are already using I2S. But well, I'm not working much on the project these days. I don't foresee the need to simulate it in Wokwi in the near future. So for me I2S doesn't really have that much priority at the moment. I don't know how much work this would be to implement.

**Comment by sutaburosu** [comment]
Yeah, better quality audio input than ADC for FFT & FastLED projects is my primary desire too. Audio output would be nice, but I don't have specific projects in mind for that. I hadn't considered that the I2S peripheral might improve FastLED's compatibility with Wokwi on ESP32 too. That would be a pleasant side-effect, if it does work.

**Comment by jackdewhurst** [comment]
Would love support for FastLED please 🙏

**Comment by urish** [comment]
The good news is that the DMA part is already complete. Leaving just the I2S part to be implemented :)

**Comment by SleepInfinity** [comment]
audio input and output would be so cool !

**Comment by baranovmv** [comment]
Both Audio input and output, via DMA
